wallets/react/src/legacy/useLegacyProviders.ts

```diff
@@ -0,0 +1,218 @@
+import type { ProviderContext, ProviderProps } from './types.js';
+import type { WalletType } from '@rango-dev/wallets-shared';
+
+import { useEffect, useReducer } from 'react';
+
+import {
+  availableWallets,
+  checkWalletProviders,
+  clearPersistance,
+  connectedWallets,
+  defaultWalletState,
+  makeEventHandler,
+  stateReducer,
+  tryPersistWallet,
+  tryRemoveWalletFromPersistance,
+} from './helpers.js';
+import { useInitializers } from './hooks.js';
+import { useAutoConnect } from './useAutoConnect.js';
+
+export function useLegacyProviders(props: ProviderProps): ProviderContext {
+  const [providersState, dispatch] = useReducer(stateReducer, {});
+
+  // Get (or add) wallet instance (`provider`s will be wrapped in a `Wallet`)
+  const getWalletInstance = useInitializers(
+    makeEventHandler(dispatch, props.onUpdateState)
+  );
+
+  // Getting providers from props and put all of them in a `Map` with an appropriate interface.
+  const listOfProviders = props.providers;
+  const wallets = checkWalletProviders(listOfProviders);
+
+  useAutoConnect({
```

Comment:
In original code, it was an useEffect (the last one in `provider.tsx`). I moved it to `useAutoConnect` and use it here.